### PR TITLE
Filled out the vertex pulling article

### PR DESCRIPTION
**Pull request by Finding-Fortune**
I recently learned vertex pulling, so I thought I would help fill out this article. The example uses bit compression, so there might be an easier example out there - hopefully not too complex.

### CODE REVIEW

File: content/wiki/vertex-pulling/index.md [hunk 1]
```diff
@@ -13,17 +13,116 @@ chapter_next = {text = "Face Pulling", link = "/wiki/face-pulling"}
 
 {% todo_notice() %} This article is a work-in-progress. {% end %}
 
-- Normal vertex-streaming does not support per-face data.
```
Comment (Longor1996) [inline content/wiki/vertex-pulling/index.md:16]: Where did this line go?

File: content/wiki/vertex-pulling/index.md [hunk 2]
```diff
@@ -13,17 +13,116 @@ chapter_next = {text = "Face Pulling", link = "/wiki/face-pulling"}
...
-- Pulling from a vertex-stream or SSBO/UBO is generally the same speed.
-- Vertex-shaders *only* have to emit a NDC-position; *how* they do so is undefined.
-- Vertices can be defined via vertex- and index-array *within* a vertex-shader.
+Traditionally, when uploading and fetching data to draw in a vertex shader, you would use a Vertex Buffer Object (VBO) linked to a Vertex Attribute Object (VAO). Optionally, an Index Buffer Object (EBO/IBO) could be attached to reuse identical vertices that overlap. For instance, on the face of a cube rendered with triangles, there are 2 triangles with 6 vertices. 2 of the vertices overlap across triangles. Hence, an index buffer can render the face with 4 indices, reusing 2 out of the 6 identical vertices.
```
Comment (Longor1996) [inline content/wiki/vertex-pulling/index.md]: I believe this needs a note that OpenGL nomenclature is being used?

**Review by Longor1996 — CHANGES_REQUESTED** [review]
Looks great! Just two notes... :'D

File: content/wiki/vertex-pulling/index.md (hunk 1 above)
Comment (Finding-Fortune) [reply content/wiki/vertex-pulling/index.md:16]: I added this line back in, in the next commit

File: content/wiki/vertex-pulling/index.md (hunk 2 above)
Comment (Finding-Fortune) [reply content/wiki/vertex-pulling/index.md]: I made a note in the next commit in bold font in the beginning that OpenGL nomenclature is used, but the same techniques are applicable in other APIs

**Review by Longor1996 — APPROVED** [review]
Nice 👍

**Comment by Finding-Fortune** [comment]
Hey @Longor1996 - I did one more quick read through and there was a small bug in the 6th step. I have a line:
`position += facePositions[currVertexID];`
but it should be:
`position += facePositions[indices[currVertexID]];`
I thought it would be excessive to make a new branch, so thought I would just ping you with this change.

**Comment by Longor1996** [comment]
Fixed with 6a2fcb0

**Pull request merged**